src/plugin.ts

```diff
@@ -24,8 +24,9 @@ async function main() {
   const utils = makeUtils(options);
   const ctx: Context = { typeMap, options, utils };
 
+  const filesToGenerate = options.emitImportedFiles ? request.protoFile : protoFilesToGenerate(request);
   const files = await Promise.all(
-    protoFilesToGenerate(request).map(async (file) => {
+    filesToGenerate.map(async (file) => {
```

Comment:
Nice!

I've been bitten by this before, but can you update `integration/codegen.ts` as well? Either with the same logic you have here, or maybe just have it always use `request.protoFile`? I don't think we have an integration test that explicitly tests that `protoFilesToGenerate` codepath, which is fine I think...

Reply:
HI!

I've updated `codegen.ts`. I think usage of `request.protoFile` should be enough fo tests.